open_humans/management/commands/vacuum_log_bucket.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime
+from urllib.parse import urlparse, parse_qs
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+import boto3
+from pyparsing import ZeroOrMore, Regex
+
+from data_import.models import DataFile, AWSDataFileAccessLog, NewDataFileAccessLog
+
+
+class Command(BaseCommand):
+    """
+    A management command vaccuming up file access logs and putting them in the database.
+    """
+
+    help = "Populates database with file access logs"
+
+    def handle(self, *args, **options):
+        self.stdout.write("Retreiving file access logs")
+        s3 = boto3.resource("s3")
+        bucket = s3.Bucket(settings.LOG_BUCKET)
+        for item in bucket.objects.all():
+            log_objects = item.get()["Body"].read().decode("utf-8").split("\n")
+            for log_entry in log_objects:
+                if not log_entry:
+                    continue
+                parser = ZeroOrMore(
+                    Regex(r"\[[^]]*\]") | Regex(r'"[^"]*"') | Regex(r"[^ ]+")
```

Comment:
Why use a regex here instead of `shlex` (which was described in the stackoverflow link I shared before)?

```
>>> import shlex
>>> teststr = 'this is a "string of words" that has some "in quotes" together'
>>> shlex.split(teststr)
['this', 'is', 'a', 'string of words', 'that', 'has', 'some', 'in quotes', 'together']
```

Reply:
I tried to get shlex to do [] as quotes, but it kept barfing on me.  This solution comes from here:
https://stackoverflow.com/questions/27303977/split-string-at-double-quotes-and-box-brackets

With shlex, I kept getting "ValueError: No closing quotation"

Pulling from the shlex documentation at [0]:

shlex.quotes    Characters that will be considered string quotes. The token accumulates until the same quote is encountered again (thus, different quote types protect each other as in the shell.) By default, includes ASCII single and double quotes.

Looking at the code of shlex (/usr/lib64/python3.x/shlex.py for me), it is not written to handle brackets-as-quotes.  So, in order to properly handle the datetime object, I had to find another solution.

I did consider just doing a log_entry.replace("[", '"').replace("]", '"'), but that seemed inelegant?  If you think doing that and using shlex is a better way, we can do that.


[0]  https://docs.python.org/3/library/shlex.html

Reply:
Oops, I see, I think I'd missed that brackets were in this as well. Thanks for the clarification!

And regex's are annoying, so it's great to document that a regex came from there (rather than you creating it from scratch) -- among other things, it means one can observe if anyone else complains it's flawed, etc etc. :) Maybe you could add a comment referencing the stackoverflow?